src/main/java/org/folio/rest/impl/InvoiceNumberImpl.java

```diff
@@ -0,0 +1,50 @@
+package org.folio.rest.impl;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.folio.rest.RestVerticle;
+import org.folio.rest.jaxrs.model.SequenceNumber;
+import org.folio.rest.jaxrs.resource.InvoiceStorageInvoiceNumber;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.tools.messages.MessageConsts;
+import org.folio.rest.tools.messages.Messages;
+import org.folio.rest.tools.utils.TenantTool;
+
+import javax.ws.rs.core.Response;
+import java.util.Map;
+
+import static io.vertx.core.Future.succeededFuture;
+import static org.folio.rest.jaxrs.resource.InvoiceStorageInvoiceNumber.GetInvoiceStorageInvoiceNumberResponse.*;
+
+public class InvoiceNumberImpl implements InvoiceStorageInvoiceNumber {
+
+  private final Messages messages = Messages.getInstance();
+
+  private static final Logger log = LoggerFactory.getLogger(InvoiceNumberImpl.class);
+  private static final String INVOICE_NUMBER_QUERY = "SELECT nextval('invoice_number')";
+
+  @Override
+  public void getInvoiceStorageInvoiceNumber(String lang, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    vertxContext.runOnContext((Void v) -> {
+      String tenantId = TenantTool.calculateTenantId(okapiHeaders.get(RestVerticle.OKAPI_HEADER_TENANT));
+      PostgresClient.getInstance(vertxContext.owner(), tenantId).selectSingle(INVOICE_NUMBER_QUERY, reply -> {
+        if (reply.succeeded()) {
+          try {
+            String invoiceNumber = reply.result().getList().get(0).toString();
+            log.debug("Retrieved invoice number: {}", invoiceNumber);
+            asyncResultHandler.handle(succeededFuture(respond200WithApplicationJson(new SequenceNumber().withSequenceNumber(invoiceNumber))));
+          } catch(Exception e) {
```

Comment:
As far as I can see the exception can happen here only if `reply.result().getList().get(0).toString()` throws exception. Not sure if this might happen...

Reply:
Yeah, and I'm not sure a 400 is appropriate in the case where a number couldn't be retrieved from the sequence either...

Maybe it makes more sense to do something like:

```
try {
  if(succeeded) { 
    ...
  } else {
    throw exception
  }
} catch (Exception e) {
  respond with 500
}
```